src/ts/order.ts

```diff
@@ -125,6 +123,28 @@ export enum OrderKind {
   BUY = "buy",
 }
 
+/**
+ * Order balance configuration.
+ */
+export enum OrderBalance {
+  /**
+   * Use ERC20 token balances.
+   */
+  ERC20 = "erc20",
+  /**
+   * Use Balancer Vault external balances.
+   *
+   * This can only be specified specified for the sell balance and allows orders
+   * to re-use Vault ERC20 allowances. When specified for the buy balance, it
+   * will be treated as {@link OrderBalance.ERC20}.
+   */
+  EXTERNAL = "external",
+  /**
+   * Use Balancer Vault external balances.
+   */
+  INTERNAL = "internal",
+}
```

Comment:
I'm not 100% happy with these string values here, and this is what the user will see when signing their order. Note that we use digests of these on the SC and encode them into 3 bits, so string length is no object.

Reply:
maybe the enum can be called OrderBalanceSource or just BalanceSource and then we can make the Vault parts explicit by vault-external, vault-internal and the erc20 one being `GPv2Allowance` or similar?

Reply:
I'm fine with this. Since its a string we can even use `Vault external` `Vault internal` `GPv2 allowance`.

Reply:
I'd like to get @anxolin's point of view as well, since ultimately we do this so that the signed data looks nicer.

Reply:
My issue with `BalanceSource` is that it is accurate for the sell token, but inaccurate for the buy token (where it is the balance "target").

Reply:
Created https://github.com/gnosis/gp-v2-contracts/issues/543 so we can merge this to continue working and don't forget to come back to this.

Reply:
Maybe BalanceTarget could be its own enum (since it only has a subset of values?). This way we could also avoid normalizing the balance for the buy side. But not sure if that makes everything else more boilerplaty

Reply:
Added this to the issue.

Reply:
Joining late for this, but I see you created an issue, so we can track it there. Thanks also for tagging!